src/components/views/avatars/BaseAvatar.tsx

```diff
@@ -20,33 +20,36 @@ limitations under the License.
 import React, { useCallback, useContext, useEffect, useState } from "react";
 import classNames from "classnames";
 import { ResizeMethod, ClientEvent } from "matrix-js-sdk/src/matrix";
+import { Avatar } from "@vector-im/compound-web";
 
-import * as AvatarLogic from "../../../Avatar";
 import SettingsStore from "../../../settings/SettingsStore";
-import AccessibleButton, { ButtonEvent } from "../elements/AccessibleButton";
+import { ButtonEvent } from "../elements/AccessibleButton";
 import RoomContext from "../../../contexts/RoomContext";
 import MatrixClientContext from "../../../contexts/MatrixClientContext";
 import { useTypedEventEmitter } from "../../../hooks/useEventEmitter";
-import { toPx } from "../../../utils/units";
 import { _t } from "../../../languageHandler";
 
 interface IProps {
-    name?: string; // The name (first initial used as default)
-    idName?: string; // ID for generating hash colours
+    name?: React.ComponentProps<typeof Avatar>["name"]; // The name (first initial used as default)
+    idName?: React.ComponentProps<typeof Avatar>["id"]; // ID for generating hash colours
     title?: string; // onHover title text
     url?: string | null; // highest priority of them all, shortcut to set in urls[0]
     urls?: string[]; // [highest_priority, ... , lowest_priority]
+    type?: React.ComponentProps<typeof Avatar>["type"];
     width: number;
+    /**
+     * @deprecated use `width` only, avatars have a 1:1 aspect ratio
+     */
     height: number;
```

Comment:
Can we just kill it? We're not a real SDK, confusion is worse than bigger diffs

Reply:
Done! I will rename `width` to `size` in a follow up PR